apps/web/vibes/soul/primitives/blog-post-card/index.tsx

```diff
@@ -16,7 +16,15 @@ export interface BlogPost {
   className?: string;
 }
 
-export function BlogPostCard({ title, image, content, href, date, author, className }: BlogPost) {
+export function BlogPostCard({
+  title,
+  image,
+  content,
+  href,
+  date,
+  author,
+  className,
+}: BlogPostCardBlogPost) {
```

Comment:
I think we should avoid having an interface drive props. Instead we should do this:

```ts
function BlogPostCard({ blogPost }: { blogPost: BlogPostCardBlogPost }) {
```

Alternatively, we could just have these be props and there's no reason to share the interface, we could share the prop interface. If you think about it from a UI perspective, the `title` doesn't necessarily have to come from a blog post—for example, in Makeswift, you could create a "fake" `BlogPostCard` that has this data provided visually.

A good example of where this breaks down is that the `id` field is required for a "blog post" but we don't want to require `id` for this component as a prop. We went through this for the `ProductCard` component